datenguide_python/query_builder.py

```diff
@@ -66,17 +100,29 @@ def __init__(
         else:
             raise TypeError("region or parent must be defined.")
```

Comment:
Why is this the case. The regions field in for an allRegions query works without specifying a parent. No parent and nuts = 1 (states) might actually be a legitimate use case.

Reply:
but then a region would be specified, right? Again, maybe we should reconsider my overall usage of Field and Query 🤔

Reply:
No the region would not be specific. From the API documentation of RegionResults  we have
regions(nuts: Int
 lau: Int
 parent: String): [Region]
page: Int
itemsPerPage: Int
total: Int

Note that in the argument list the is not a single "!" for any argument. This means all arguments are optional. A query of the form
allRegions {
    regions {
        name
        ....
    }
}

would be valid. This particular query is however not so useful as it would return too many results. A query of the form 
allRegions {
    regions (nuts : 1) {
        name
        ....
    }
}
however is still valid although it does **not have a parent id** would return reasonable results because it only outputs statistics for all states, which is manageble amout and actually also something that one might want to compare.